_posts/zh/newsletters/2019-05-29-newsletter.md

```diff
@@ -0,0 +1,179 @@
+
+- [LND #3098][] 增加了守护进程等待由远程对等节点启动的通道资金交易确认的最大块数，从 288 块（两天）增加到 2,016 块（两周）。这允许有耐心的用户支付更低的交易费用。
+
+- [C-Lightning #2647][] 指定了一个默认的插件目录，即使未指定 `--plugin` 或 `--plugin-dir` 配置参数，也会从该目录自动加载插件。目前这是闪电守护进程配置目录中的 `plugins` 目录。
+
+- [C-Lightning #2650][] 添加了一个新的插件挂钩，当远程对等节点尝试与本地节点打开通道时触发。这样插件可以在通道打开之前拒绝通道打开或执行其他操作。
+
+- [Eclair #952][] 添加了一个 `sendtoroute` 方法，允许用户手动选择初始路由付款的通道。这可以让他们选择哪些通道的资金被耗尽。
+
+- [Eclair #965][] 允许用户在创建发票时指定 preimage。这可以用于安全生成不可猜测的发票标识符的系统，例如[原子交换][atomic swap]或一组合同条款与 [pay-to-contract 格式][p2c] 中的 nonce 结合使用。
+
+## 新出版计划
+
+从本周开始，Optech Newsletter 将改为每周三发布，而不是每周二。这将给我们额外的一个工作日来审查和编辑 Newsletter 草稿。
+
+## 特别感谢
+
+我们感谢 Jeremy Rubin 和 Anthony Towns 对本 Newsletter 草稿的审查，包括向我们描述输出树的想法。我们还感谢 Pieter Wuille 帮助我们更好地理解使用 MuSig 聚合密钥和签名时的互动要求。本 Newsletter 发布版本中的任何错误均为作者的责任。
+
+## 脚注
+
+[^legacy-cleanstack]:
+    Segwit v0（P2WSH）和 Tapscript 提案要求最终堆栈仅包含一个*真*元素才能成功。这称为*清洁堆栈*规则。裸输出和 P2SH 输出的传统脚本允许堆栈包含多个项目，并在堆栈顶部项目为*真*时成功。然而，不具有清洁堆栈的传统交易将不会被 Bitcoin Core 的默认内存池策略中继或挖掘。清洁堆栈规则有助于减少交易的可塑性，因为对 scriptSig 或见证的任何额外或删除操作都会改变交易的费用率以及（对于传统交易）其 txid。
+
+[^taproot-pubkeys]:
+    提议的 Taproot 地址格式（v1 segwit 地址）直接在地址中包含一个公钥，因此任何拥有一组 Taproot 地址的人都可以使用它们创建一个聚合公钥。然而，一些用户可能会使用没有人拥有（或能合理生成）相应私钥的公钥来创建 Taproot 地址。因此，任何创建聚合公钥的人可能不应该假设 Taproot 地址本身就是公钥，应该收集单独的公钥。此外，最好不要在比特币的多个地方重用同一个公钥。为了简化对 COSHV 的描述，我们在本 Newsletter 的示例中省略了收集公钥的额外步骤。在您实现从本 Newsletter 或互联网上任何其他地方读到的协议之前，请考虑咨询比特币专家。
+
+{% include linkers/issues.md issues="3098,2647,2650,952,965" %}
+[channel factories]: https://www.tik.ee.ethz.ch/file/a20a865ce40d40c8f942cf206a7cba96/Scalable_Funding_Of_Blockchain_Micropayment_Networks.pdf
+[p2c]: https://arxiv.org/abs/1212.3257
+[atomic swap]: https://en.bitcoin.it/wiki/Atomic_swap
+[emptystack1]: https://lists.linuxfoundation.org/pipermail/bitcoin-dev/2019-May/016935.html
+[emptystack0]: https://lists.linuxfoundation.org/pipermail/bitcoin-dev/2018-December/016558.html
+[emptystack reply]: https://lists.linuxfoundation.org/pipermail/bitcoin-dev/2019-May/016947.html
+[elliptic curve]: https://en.wikipedia.org/wiki/Elliptic_curve
+[smaller v1 spk]: https://lists.linuxfoundation.org/pipermail/bitcoin-dev/2019-May/016943.html
+[vid return to fees]: https://www.youtube.com/watch?v=ihUQ4C42KUk
+[coshv flex]: https://lists.linuxfoundation.org/pipermail/bitcoin-dev/2019-May/016936.html
+[elementsproject.org]: https://elementsproject.org/
+[blockstream liquid]: https://blockstream.com/liquid/
+[catenates]: https://en.wiktionary.org/wiki/catenate
+[uncompressed public keys]: https://btcinformation.org/en/developer-guide#public-key-formats
+[ivgi tweet]: https://twitter.com/shesek/status/1131733590235131905
+[updated qr paragraph]: /zh/bech32-sending-support/#qrcode-edit
+[optech twitter]: https://twitter.com/bitcoinoptech
+[optech contributors]: /about/#contributors
+[special section about the proposal]: #proposed-transaction-output-commitments
```

Comment:
Needs to be updated to the current header for the 'Proposed transaction output commitments' section that covers OP_CHECKOUTPUTSHASHVERIFY

Reply:
Initially, I did not recognize this point, but I have added links internally to correct the current content. I have noticed similar issues in subsequent content and will pay attention to them in future work.